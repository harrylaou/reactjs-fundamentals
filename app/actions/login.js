export const LOGIN_USER_OK = 'LOGIN_USER_OK'
export const LOGOUT_USER = 'LOGOUT_USER'

export const loginUser = () => ({
  type: LOGIN_USER_OK
})

export const logoutUser = () => ({
  type: LOGOUT_USER
})