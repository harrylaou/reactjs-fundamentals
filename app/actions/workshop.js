export const RECEIVE_WORKSHOP = 'RECEIVE_WORKSHOP'


export const receiveWorkshop = workshop => ({
  type: RECEIVE_WORKSHOP,
  workshop
})