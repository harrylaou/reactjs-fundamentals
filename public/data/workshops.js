[
  {
    "title": "React Router and fetch data",
    "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
    "students" : "Unlimited",
    "price": "Free",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        }
    ]
  },
  {
    "title": "React Master class",
    "url": "http://techdency.com/#react-training",
    "price": "Â£ 595",
    "students" : "10 per instructor",
    "instructors": [
        {
            "name":"Alex Lobera",
            "url": "http://alexlobera.com/",
            "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses.jpg"
        },
        {
            "name":"Davy Engone",
            "url": "https://github.com/davyengone",
            "avatar": "https://pbs.twimg.com/profile_images/589582545022545921/qQRHggUr.jpg"
        }
    ]
  }
]