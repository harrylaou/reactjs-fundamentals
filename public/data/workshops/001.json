{
  "id": "001",
  "title": "React Router and fetch data",
  "url": "http://www.meetup.com/JS-Lab-London/events/231907561/",
  "students" : "Unlimited",
  "price": "Free",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    }
  ]
}