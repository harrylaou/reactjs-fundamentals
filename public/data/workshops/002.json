{
  "id": "002",
  "title": "React Master class",
  "url": "http://techdency.com/#react-training",
  "price": "Â£ 595",
  "students" : "10 per instructor",
  "instructors": [
    {
      "name":"Alex Lobera",
      "url": "http://alexlobera.com/",
      "avatar": "https://pbs.twimg.com/profile_images/1540939151/whereIsAlexWithGlasses_400x400.jpg"
    },
    {
      "name":"Mark Rivelles",
      "url": "https://github.com/fsdevlondon",
      "avatar": "https://pbs.twimg.com/profile_images/784142672370003969/Hj51EXIt.jpg"
    }
  ]
}