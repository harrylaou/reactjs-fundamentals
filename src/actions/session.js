export const LOG_OUT = 'LOG_OUT'
export const LOGGED_IN = 'LOGGED_IN'

export const logout = () => ({
  type: LOG_OUT
})

export const loggedIn = token => ({
  type: LOGGED_IN,
  token
})
