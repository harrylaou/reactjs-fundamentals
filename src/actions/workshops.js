export const RECEIVE_WORKSHOPS = 'RECEIVE_WORKSHOPS'

export const receiveWorkshops = workshops => ({
  type: RECEIVE_WORKSHOPS,
  workshops
})
