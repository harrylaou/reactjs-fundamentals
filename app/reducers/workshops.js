import {RECEIVE_WORKSHOPS} from '../actions/workshops'

const workshops = (state = [], action) => {
  switch (action.type) {
    case RECEIVE_WORKSHOPS: {
      return action.workshops
    }
    default:
      return state
  }
}

export default workshops
