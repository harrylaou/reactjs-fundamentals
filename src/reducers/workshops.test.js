import workshops from './workshops'
import {RECEIVE_WORKSHOPS} from '../actions/workshops'

describe('#Reducer: Workshop', () => {
  it('Return default state',() => {
    expect(workshops(undefined, { type: 'default_action' })).toEqual([])
    expect(workshops(undefined, { type: 'default_action' })).toMatchSnapshot()
  })

  it('Return state',() => {
    expect(workshops(["data"], { type: 'another_action' })).toEqual(["data"])
    expect(workshops(["data"], { type: 'another_action' })).toMatchSnapshot()
  })

  it(`When Action ${RECEIVE_WORKSHOPS} return state`,() => {
    expect(workshops(["meetup"], { type: RECEIVE_WORKSHOPS, workshops: ["React Fundamentals"] })).toEqual(["meetup", "React Fundamentals"])
    expect(workshops(["meetup"], { type: RECEIVE_WORKSHOPS, workshops: ["React Fundamentals"] })).toMatchSnapshot()
  })
})