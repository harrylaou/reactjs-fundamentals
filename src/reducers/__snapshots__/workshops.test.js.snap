// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Reducer: Workshop Return default state 1`] = `Array []`;

exports[`#Reducer: Workshop Return state 1`] = `
Array [
  "data",
]
`;

exports[`#Reducer: Workshop When Action RECEIVE_WORKSHOPS return state 1`] = `
Array [
  "React Fundamentals",
]
`;
